itests/manual_cc_test.go

```diff
@@ -0,0 +1,216 @@
+	req.NoError(err)
+	t.Logf("SectorOnChainInfo %d: %+v", bSectorNum, soi)
+
+	sp, err := client.StateSectorPartition(ctx, maddrB, bSectorNum, r.TipSet)
+	req.NoError(err)
+	t.Logf("SectorPartition %d: %+v", bSectorNum, sp)
+	bSectorDeadline := sp.Deadline
+	bSectorPartition := sp.Partition
+
+	p, err = client.StateMinerPower(ctx, maddrB, r.TipSet)
+	req.NoError(err)
+	t.Logf("MinerB RBP: %v, QaP: %v", p.MinerPower.QualityAdjPower.String(), p.MinerPower.RawBytePower.String())
+	require.True(t, p.MinerPower.RawBytePower.IsZero())
+
+	di, err := client.StateMinerProvingDeadline(ctx, maddrB, types.EmptyTSK)
+	req.NoError(err)
+	t.Logf("MinerB Deadline Info: %+v", di)
+
+	// Use the current deadline to work out when the deadline we care about (bSectorDeadline) is open
+	// and ready to receive posts
+	deadlineCount := di.WPoStPeriodDeadlines
+	epochsPerDeadline := uint64(di.WPoStChallengeWindow)
+	currentDeadline := di.Index
+	currentDeadlineStart := di.Open
+	waitTillEpoch := abi.ChainEpoch((deadlineCount-currentDeadline+bSectorDeadline)*epochsPerDeadline) + currentDeadlineStart - di.WPoStProvingPeriod + 1
+
+	t.Logf("Waiting %d until epoch %d to get to deadline %d", waitTillEpoch-di.CurrentEpoch, waitTillEpoch, bSectorDeadline)
+	client.WaitTillChain(ctx, kit.HeightAtLeast(waitTillEpoch))
+
+	// We should be up to the deadline we care about
+	di, err = client.StateMinerProvingDeadline(ctx, maddrB, types.EmptyTSK)
+	req.NoError(err)
+	req.Equal(di.Index, bSectorDeadline, "should be in the deadline of the sector to prove")
+
+	t.Log("Submitting WindowedPoSt...")
+
+	head, err = client.ChainHead(ctx)
+	req.NoError(err)
+	rand, err := client.StateGetRandomnessFromTickets(ctx, crypto.DomainSeparationTag_PoStChainCommit, di.Open, nil, head.Key())
+	require.NoError(t, err)
+
+	postParams := miner.SubmitWindowedPoStParams{
+		ChainCommitEpoch: di.Open,
+		ChainCommitRand:  rand,
+		Deadline:         bSectorDeadline,
+		Partitions:       []miner.PoStPartition{{Index: bSectorPartition}},
+		Proofs: []proof.PoStProof{{
+			PoStProof:  minerBInfo.WindowPoStProofType,
+			ProofBytes: []byte{0x1, 0x2, 0x3, 0x4},
+		}},
+	}
+
+	enc = new(bytes.Buffer)
+	req.NoError(postParams.MarshalCBOR(enc))
+
+	m, err = client.MpoolPushMessage(ctx, &types.Message{
+		To:     maddrB,
+		From:   minerB.OwnerKey.Address,
+		Value:  types.NewInt(0),
+		Method: builtin.MethodsMiner.SubmitWindowedPoSt,
+		Params: enc.Bytes(),
+	}, nil)
+	req.NoError(err)
+
+	r, err = client.StateWaitMsg(ctx, m.Cid(), 2, api.LookbackNoLimit, true)
+	req.NoError(err)
+	require.True(t, r.Receipt.ExitCode.IsSuccess())
+
+	t.Log("Checking power after PoSt...")
+
+	p, err = client.StateMinerPower(ctx, maddrB, r.TipSet)
+	req.NoError(err)
+	// require.True(t, p.MinerPower.RawBytePower.IsZero())
+	t.Logf("MinerB RBP: %v, QaP: %v", p.MinerPower.QualityAdjPower.String(), p.MinerPower.RawBytePower.String())
+	req.Equal(uint64(2<<10), p.MinerPower.RawBytePower.Uint64())    // 2kiB RBP
```

Comment:
Do you start getting power only after submitting the first WindowPost ?

Reply:
Yes, I believe this is how the protocol is intended to work - precommit, commit, then prove before you get power, and we have to wait for the right window to submit the proof, hence the complicated deadline wait logic.